cli/src/commands/branch.rs

```diff
@@ -760,9 +760,13 @@ fn cmd_branch_list(
                 let remote_added_ids = remote_ref.target.added_ids().cloned().collect_vec();
                 let local_added_ids = local_target.added_ids().cloned().collect_vec();
                 let remote_ahead_count =
-                    revset::walk_revs(repo.as_ref(), &remote_added_ids, &local_added_ids)?.count();
+                    revset::walk_revs(repo.as_ref(), &remote_added_ids, &local_added_ids)?
+                        .iter()
+                        .count();
```

Comment:
If anyone had implemented `Revset::count()` efficiently, they would see a performance regression with this commit. I think it's pretty safe to say that no one has (context for others: at Google it just panics, which is why Jonathan is working on this), so it doesn't matter in practice. I.e., if there had been other projects using the jj library, then I think we should have squashed the next commit into this one to prevent even that temporary performance regression (I think temporary performance regressions can be okay where they're harder to avoid).

Reply:
I've gone ahead and squashed this.